design/nfs.md

```diff
@@ -0,0 +1,157 @@
+# Add NFS to Rook
+
+## Overview
+
+Rook is an open source orchestrator for distributed storage systems running in kubernetes, currently in alpha state and has focused initially on orchestrating Ceph on top of Kubernetes. This document explores a design to add NFS to Rook. This is a part of the rook feature request [#1551](https://github.com/rook/rook/issues/1502).
+
+## Rook Architecture
+
+Rook turns distributed storage software into a self-managing, self-scaling, and self-healing storage services. It does this by automating deployment, bootstrapping, configuration, provisioning, scaling, upgrading, migration, disaster recovery, monitoring, and resource management. Rook uses the facilities provided by the underlying cloud-native container management, scheduling and orchestration platform to perform its duties.
+![Rook Architecture on Kubernetes](../Documentation/media/rook-architecture.png)
+
+## Network File System (NFS)
+
+NFS allows remote hosts to mount file systems over a network and interact with those file systems as though they are mounted locally. This enables system administrators to consolidate resources onto centralized servers on the network.
+
+## Why NFS?
+
+NFS is widely used for persistent storage in kubernetes cluster. Using NFS storage is a convenient and easy way to provision storage for applications.
+An NFS volume allows an existing NFS (Network File System) share to be mounted into the pod.
+The contents of an NFS volume are preserved and the volume is merely unmounted if the pod is stopped/destroyed. This means that an NFS volume can be pre-populated with data, and that data can be “handed off” between pods.
+NFS supports multiple read/write simultaneously so a single share can be attached to multiple pods.
+
+## Design
+
+Rook brings storage platforms to Kubernetes with a fully Kubernetes-native design. It integrates deeply into cloud native environments leveraging extension points and providing a seamless experience for scheduling, lifecycle management, resource management, security, monitoring, and user experience.
```

Comment:
It seems you have already mentioned the overview in the `Rook architecture` section, so maybe you don't need this line here?